src/other/Slider.js

```diff
@@ -223,17 +238,23 @@
             .render()
         ;
 
-        this.xScale
-            .domain([this.low(), this.high()])
-            .range([-width / 2, width / 2 - (this.showPlay() ? this.loopDiameter() + this.loopGutter() + this.playDiameter() + this.playGutter() : 0)])
-        ;
-
         if ((this.high() - this.low()) / this.step() <= 10) {
             this.axis.tickValues(d3.merge([d3.range(this.low(), this.high(), this.step()), [this.high()]]));
         } else {
-            this.axis.tickValues(null);
+            this.axis.tickValues(d3.merge([d3.range(this.low(), this.high(), (this.high() - this.low()) / this.step()), [this.high()]]));
         }
+
+        width -= this.showPlay() ? this.loopDiameter() + this.loopGutter() + this.playDiameter() + this.playGutter() : 0;
+        this.xScale
+            .domain([this.low(), this.high()])
+            .range([leftPos, leftPos + width])
+        ;
+        var delta = this.calcDelta(domNode, element, leftPos, width);
+        this.xScale
+            .range([leftPos - delta.left, leftPos + width - delta.right])
+        ;
         this.axisElement
+            .attr("transform", "translate(0, -10)")
```

Comment:
You need to calculate the height - what happens if someone is using a really big font?